search-service/src/main/kotlin/com/egm/stellio/search/csr/service/ContextSourceUtils.kt

```diff
@@ -0,0 +1,177 @@
+package com.egm.stellio.search.csr.service
+
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.raise.either
+import arrow.core.right
+import com.egm.stellio.search.csr.model.*
+import com.egm.stellio.shared.model.CompactedAttributeInstance
+import com.egm.stellio.shared.model.CompactedAttributeInstances
+import com.egm.stellio.shared.model.CompactedEntity
+import com.egm.stellio.shared.model.InternalErrorException
+import com.egm.stellio.shared.util.*
+import com.egm.stellio.shared.util.JsonLdUtils.JSONLD_CONTEXT
+import com.egm.stellio.shared.util.JsonLdUtils.JSONLD_ID_TERM
+import com.egm.stellio.shared.util.JsonLdUtils.JSONLD_TYPE_TERM
+import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_DATASET_ID_TERM
+import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_MODIFIED_AT_TERM
+import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_OBSERVED_AT_TERM
+import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_SCOPE_TERM
+import com.egm.stellio.shared.util.JsonLdUtils.logger
+import com.fasterxml.jackson.core.JacksonException
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpStatus
+import org.springframework.util.MultiValueMap
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientException
+import org.springframework.web.reactive.function.client.awaitBody
+import org.springframework.web.reactive.function.client.awaitExchange
+import java.net.URI
+import java.time.ZonedDateTime
+import kotlin.random.Random.Default.nextBoolean
+
+typealias CompactedEntityWithMode = Pair<CompactedEntity, Mode>
+
+object ContextSourceUtils {
+
+    suspend fun getDistributedInformation(
+        httpHeaders: HttpHeaders,
+        csr: ContextSourceRegistration,
+        path: String,
+        params: MultiValueMap<String, String>
+    ): Either<NGSILDWarning?, CompactedEntity> = either {
```

Comment:
the correct return type should be `Either<NGSILDWarning, CompactedEntity?>` (as "responding with no data and the HTTP status code 404 - Not Found should not be considered as abnormal behaviour for distributed operations")